api/apps/api/src/modules/scenarios/planning-units/scenario-planning-units-protection-status-calculator-service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+
+import { DbConnections } from '@marxan-api/ormconfig.connections';
+import { ScenariosPlanningUnitGeoEntity } from '@marxan/scenarios-planning-unit';
+import { Scenario } from '../scenario.api.entity';
+import { isNil } from 'lodash';
+
+@Injectable()
+export class ScenarioPlanningUnitsProtectedStatusCalculatorService {
+  constructor(
+    @InjectRepository(
+      ScenariosPlanningUnitGeoEntity,
+      DbConnections.geoprocessingDB,
+    )
+    private readonly puRepo: Repository<ScenariosPlanningUnitGeoEntity>,
+  ) {}
+
+  /**
+   * Calculate default lock status of planning units given their intersection
+   * with protected areas.
+   */
+  async calculatedProtectionStatusForPlanningUnitsIn(scenario: Scenario): Promise<void> {
```

Comment:
tests would be great addition (for both logic /unit/ and e2e/integration to verify it actually changes desired columns.

Reply:
another item for second pass, sorry 😢